classes/Content/class.xudfContentGUI.php

```
@@ -165,47 +174,51 @@ protected function checkAndSendNotification(): void
 
     protected function returnToParent(): void
     {
-        $this->dic->ctrl()->setParameterByClass(ilRepositoryGUI::class, 'ref_id', $this->tree->getParentId((int) $_GET['ref_id']));
+        $refId = $this->httpWrapper->query()->retrieve(
+            "ref_id",
+            $this->refinery->kindlyTo()->int()
```

If you require this parameter, you don't need to use the "kindlyTo" refinery, but can use the normal one.

If it is not there, this should throw an error here and not somewhere later where it's used (Im aware that just 1 line difference here, but its about the pattern).

---

Alternatively you should provide a fallback on the usage if you cant ensure this paramter is set.

---

Using that throws an exception since ids in the query are not of type integer,

kindlyTo turns strings into integer.
to expects it to already be an integer

---

I'm sorry I got that mixed up with another refinery-based issue :sweat_smile: 
Of course you are right, that correct this way.